File: src/main/java/dev/kikugie/worldrenderer/property/RenderPropertyBundle.java
package dev.kikugie.worldrenderer.property;

import net.minecraft.client.util.math.MatrixStack;

public interface RenderPropertyBundle {
    DoubleProperty scale();

    IntegerProperty rotation();

    IntegerProperty slant();

    IntegerProperty lightAngle();

    IntegerProperty xOffset();

    IntegerProperty yOffset();

    void applyToViewMatrix(MatrixStack modelViewStack);
}

File: src/main/kotlin/dev/kikugie/worldrenderer/property/RenderPropertyBundle.kt
package dev.kikugie.worldrenderer.property

import net.minecraft.client.util.math.MatrixStack

interface RenderPropertyBundle {
    var xOffset: Int
    var yOffset: Int

    var scale: Double
    var rotation: Double
    var slant: Double
    var lightAngle: Double

    fun applyToViewMatrix(matrices: MatrixStack)
}